nitro-protocol/contracts/libraries/ECRecovery.sol

```diff
@@ -0,0 +1,28 @@
+//SPDX-License-Identifier: MIT License
+pragma solidity 0.7.4;
+
+library ECRecovery {
+
+  /**
+   * @dev Recover signer address from a message by using his signature
+   * @param hash bytes32 message, the hash is the signed message. What is recovered is the signer address.
+   * @param v v part of a signature
+   * @param r r part of a signature
+   * @param s s part of a signature
+   */
+  function recover(bytes32 hash, uint8 v, bytes32 r, bytes32 s) internal pure returns (address) {
+
+    // Version of signature should be 27 or 28, but 0 and 1 are also possible versions
+    if (v < 27) {
+      v += 27;
+    }
+
+    // If the version is correct return the signer address
+    if (v != 27 && v != 28) {
+      return (address(0));
+    } else {
+      return ecrecover(hash, v, r, s);
+    }
+  }
+
+}
```

Comment:
This file is new, but doesn't appear to be used anywhere? 

It looks to me like a wrapper for `ecrecover` which may add some extra safety. Is that right? Is there something equivalent in the openzeppelin library that we could use https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/ECDSA.sol ?

Reply:
It is a wrapper for `ecrecover` indeed. The reason this file was introduced is inconsistency of returned `v` signature value in `go-ethereum` (as described [here](https://github.com/ethereum/go-ethereum/issues/19751)) and its difference with what pure `ecrecover` solidity function [expects](https://bitcoin.stackexchange.com/a/112489) (i.e. `v` to be either 27 or 28).

We thought of using fault-proof OpenZeppelin analogy, but it has a lot of unnecessary tweaks and, more importantly, [requires modification](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/ECDSA.sol#L159-L160) to support what we need. Thus it was simpler and clearer to make a library with the same modifications that should have been done to OpenZeppelin contract.

Reply:
I see. I wonder if it is better to fix this problem in the `go-nitro` client code, rather than on chain? That seems to be the advice from the `go-ethereum` team. What do you think?

Reply:
I agree. We will try to fix it from `go-nitro` client code ASAP.